src/ct_eval.cpp

```diff
@@ -86,7 +86,7 @@ TypedAST::IfElseStatement* ct_eval(
 
 TypedAST::ForStatement* ct_eval(
     TypedAST::ForStatement* ast, TypeChecker& tc, TypedAST::Allocator& alloc) {
-	ast->m_declaration = ct_eval(ast->m_declaration, tc, alloc);
+	ct_eval(&ast->m_declaration, tc, alloc);
```

Comment:
we need to keep this in mind, in case `ct_eval` starts doing more complicated things.